### PR TITLE
feat!: Making Datastore implement Closeable

### PR DESCRIPTION
**Pull request by jainsahab**
Noticed the below error in some scenarios
image

With gRPC it has become important to close all the `ManagedChannel`.

### CODE REVIEW

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/spi/v1/DatastoreRpc.java [hunk 1]
```diff
@@ -96,4 +96,9 @@ BeginTransactionResponse beginTransaction(BeginTransactionRequest request)
   default RunAggregationQueryResponse runAggregationQuery(RunAggregationQueryRequest request) {
     throw new UnsupportedOperationException("Not implemented.");
   }
+
+  @Override
+  default void close() throws Exception {
```
Comment (kolea2) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/spi/v1/DatastoreRpc.java:101]: why make this a default method? Should we just implement in all places?

**Review by kolea2 — CHANGES_REQUESTED** [review]
two general todos:

1. Would be nice to refactor the samples to use a try-with-resources now that this is added. Can you file an issue to track?
2. Maybe add a small test to ensure the client is closed

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/spi/v1/DatastoreRpc.java (hunk 1 above)
Comment (jainsahab) [reply google-cloud-datastore/src/main/java/com/google/cloud/datastore/spi/v1/DatastoreRpc.java:101]: We can do that, but they will be empty too like this one.

Comment (jainsahab) [reply google-cloud-datastore/src/main/java/com/google/cloud/datastore/spi/v1/DatastoreRpc.java]: addressed in 4ed8259.

**Comment by jainsahab** [comment]
> two general todos:
> 
> 1. Would be nice to refactor the samples to use a try-with-resources now that this is added. Can you file an issue to track?
> 2. Maybe add a small test to ensure the client is closed

For 1, #1248  is created
For 2, added a test in the latest snapshot.

**Review by kolea2 — APPROVED** [review]
lgtm with one question

**Comment by suztomo** [comment]
The "!" in the pull request. Would you add the description why it's breaking change and why it's ok for library users?

**Comment by suztomo** [comment]
I now know the destination branch is "V3-experimental".

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/RetryAndTraceDatastoreRpcDecorator.java [hunk 2]
```diff
@@ -109,6 +109,16 @@ public RunAggregationQueryResponse runAggregationQuery(RunAggregationQueryReques
         () -> datastoreRpc.runAggregationQuery(request), SPAN_NAME_RUN_AGGREGATION_QUERY);
   }
 
+  @Override
+  public void close() throws Exception {
+    datastoreRpc.close();
```
[the remark this answers is not in the document]
Comment (jainsahab) [reply google-cloud-datastore/src/main/java/com/google/cloud/datastore/RetryAndTraceDatastoreRpcDecorator.java:114]: `DatastoreImpl` will catch the exception generated here and print the stack trace.

**Pull request merged**